### ISSUE TITLE
Submitting Restaurant assignment

### ISSUE THREAD

**Issue by knguyen88** [knguyen88/coderschool-restaurant#1]
Here is my restaurant assignment work.

/cc @coderschoolreview

**Comment by harley** [comment]
Hi @knguyen88

Great job :+1: The goal of this homework is to understand routes (RESTful and custom), use MVC correctly for simple pages, generate simple test data, use validations, use ActiveRecord for filtering, sorting and association.

The heroku page looks a little messy but I looked into your code to see a lot of good things:
- Clean routes
- Models: Nice use of enum, has_many through, attr_accessor, after_create
- Using the latest ruby 2.3 and your'e using RVM's .ruby-version and .ruby-gemset!
- Nice job with the `db/seeds.rb` file
- Great job making `MenuItems#index` handle various search and sorting options. It can be a lot cleaner if you refactor the code into your MenuItem model, under a main method called `def self.search(query)` or something
- Nice job with a little touch of ajax.

### Suggestions:
- Note that `$(document).ready(` does not work well with turbolinks. See the 5.2 Page Change Events section in this guide for more information: http://guides.rubyonrails.org/working_with_javascript_in_rails.html#how-turbolinks-works
- Your visual layout could be better (take advantage of columns, colors, mouse effect, jquery effect etc)
- After I search for a product, you should keep the search text in the search input
- It's not clear that I can click on each row in the menu item table. Use css to set `cursor: pointer` for the `tr` elements to make it clearer.
- After I click on a food item, clicking on the Order button adds it to the Order table above but it is VERY hard to notice.
- It's interesting that you use bootstrap 3 instead of bootstrap 4. I liked the bootstrap-sass gem too. The bootstrap 4 gem is still buggy. Note that there are several breaking changes when you switch to bootstrap 4 later (for example, they replace `panel` and a few other elements with `card`)
- Should add **validations** for your models, especially `MenuItem` and `OrderItem`!!
- No way for me to add/remove menu items
- Consider displaying the total price under the Order table.
- Use `.panel-body` for your panel inside Delivery to fix the padding issue ![](https://dl.dropboxusercontent.com/spa/vkwvskbavc27sn4/-9h6alkb.png)
- In `Order#total` method, this piece of code `menu_items.inject(0) { |total, item| total + item.price }` can be shortened using the `.sum` method. You can use `menu_items.sum(:price)` which uses PostgreSQL to sum the price for you. You can also use `menu_items.to_a.sum(&:price)` which converts `menu_items` to an array and then use the `Array#sum` method.

Overall, this is a good submission in terms of functionality. For future submissions, I encourage you to spend just a little more time to make the UI/UX a little more pleasing and obvious to the user. Great to see that you're already familiar with many key Rails concepts. Rock on :rocket:
